### PR TITLE
Interpolate I8 nearest planar

### PR DESCRIPTION
**Pull request by apavliuk-altran**
### Details:
 - update for Interpolate nearest planar kernel to support int8/uint8 precision without fusing and type conversion; is enabled for AVX2 and SSE41 instruction sets.
 - update for Interpolate nearest planar kernel to support fp32 precision with fusing and type conversion; is enabled for SSE41 instruction set
 - new Interpolate CPU function tests which cover these kernels

### Tickets:
 - [CVS-55387]

### CODE REVIEW

**Comment by apavliuk-altran** [comment]
@lohika-denis-kotov, please take a look

**Comment by dmitry-gorokhov** [comment]
@chenhu-wang Please take a look

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp
Comment (chenhu-wang) [inline src/plugins/intel_cpu/src/nodes/interpolate.cpp:303]: Are you able to emulate with simple mov() based on top of stack pointed by rsp. _extrd() and _insrb() have relative larger and varied latency on different platforms. We have similar gather emulation implementation in interpolate cubic mode. You can ref here:https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/nodes/interpolate.cpp#L1286-L1309

Comment (chenhu-wang) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp:303]: Use uni_xxx(such as uni_vpinsrb()) version as much as possible.

1. Run vpinsrb() on sse41 will have illegal instruction error.

2. Run pinsrb() on avx2/avx512 have performance penalty sometimes. 

use uni_xxx can help you handle this.

Comment (apavliuk-altran) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp]: @chenhu-wang, Changed gather as you suggested, it gained some performance.

**Comment by dmitry-gorokhov** [comment]
@chenhu-wang Do you have any additional comments?

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp
Comment (chenhu-wang) [inline src/plugins/intel_cpu/src/nodes/interpolate.cpp:1352]:
```suggestion
    inline void gather_i8_i32_indices_store(const Xbyak::Reg64 &dest_addr, const Xbyak::Reg64 &source_addr,
                                            const Xbyak::Reg64 &ind_addr, int gather_num) {
```
let support this signature to optimize tails process.

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp
Comment (chenhu-wang) [inline src/plugins/intel_cpu/src/nodes/interpolate.cpp:389]: for all the tails, we can handle it in one vector process, without loop. Gather all the tails to vector register + post_ops + store with store_number parameter. To support this, we need emu_gather() with gather_num parameter.

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp
Comment (chenhu-wang) [inline src/plugins/intel_cpu/src/nodes/interpolate.cpp:303]: I believe we can support u8 as well with current implementation. Just relax conditions.

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp
Comment (chenhu-wang) [inline src/plugins/intel_cpu/src/nodes/interpolate.cpp:351]: Report exceptions will just lead some models fail to run, we need manually handle it to run successfully. Two options:

1. Disable fusion directly for this case in [canBeFuse()](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/nodes/interpolate.cpp#L3228). Then there will be no post-ops in this case, and post-ops is processed in another node successfully.
2. Support fusion in all cases. If there are post-ops, we need gather u8/i8(src_prc) data to FP32 in register(need precision convert), apply post-ops fusion, store FP32 to dst_prc with store_num. In this case, vector step is always (vlen / sizeof(float));

@dmitry-gorokhov , @ceciliapeng2011 what do you think which one is preferred?

**Review by chenhu-wang — CHANGES_REQUESTED** [review]

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp
Comment (chenhu-wang) [inline src/plugins/intel_cpu/src/nodes/interpolate.cpp:1356]: Let remove these lines, I believe it's safe in current code path.

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp
Comment (chenhu-wang) [inline src/plugins/intel_cpu/src/nodes/interpolate.cpp:1367]: Let remove this 2 lines.

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp
Comment (chenhu-wang) [inline src/plugins/intel_cpu/src/nodes/interpolate.cpp:1828]: why not support sse41 and avx512 target, the emulation should and does not have such limitations.

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp
Comment (chenhu-wang) [inline src/plugins/intel_cpu/src/nodes/interpolate.cpp:1829]: extend to U8 as well.

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp
Comment (chenhu-wang) [inline src/plugins/intel_cpu/src/nodes/interpolate.cpp:1937]: could you please extend ISA and precisions.

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp
Comment (chenhu-wang) [inline src/plugins/intel_cpu/src/nodes/interpolate.cpp:3196]:
```suggestion
    } else if (mayiuse(cpu::x64::avx2) && allowFp32) {
        interpolateKernel.reset(new jit_uni_interpolate_kernel_f32_i8<cpu::x64::avx2>(jcp, *attr.get()));
    } else if (allowI8) {
        if (mayiuse(cpu::x64::avx512_core)) {
            interpolateKernel.reset(new jit_uni_interpolate_kernel_f32_i8<cpu::x64::avx512_core>(jcp, *attr.get()));
        } else if (cpu::x64::avx2) {
            interpolateKernel.reset(new jit_uni_interpolate_kernel_f32_i8<cpu::x64::avx2>(jcp, *attr.get()));
        } else if (cpu::x64::sse41) {
           interpolateKernel.reset(new jit_uni_interpolate_kernel_f32_i8<cpu::x64::sse41>(jcp, *attr.get()));
        }
    }
```

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp
Comment (apavliuk-altran) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp]: Done

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp
Comment (apavliuk-altran) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp]: Done

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp
Comment (apavliuk-altran) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp]: Done

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp
Comment (apavliuk-altran) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp]: Done

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp
Comment (apavliuk-altran) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp]: Done

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp
Comment (apavliuk-altran) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp]: Done

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp
Comment (apavliuk-altran) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp]: Done

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp
Comment (apavliuk-altran) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp]: Done

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp
Comment (apavliuk-altran) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp]: For now, added a check to `Interpolate::canFuse()`

**Comment by apavliuk-altran** [comment]
Updated the PR to support U8 precision and all ISA for I8/U8 precisions.
After running the benchmarks for various shapes and Interpolate params, added a Threshold table with thresholds for blocked and channel-first layouts.

**Comment by apavliuk-altran** [comment]
Left threshold for channel first and blocked layouts as 1 according to the new benchmarks and suggestion from @chenhu-wang.

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp
Comment (chenhu-wang) [inline src/plugins/intel_cpu/src/nodes/interpolate.cpp:3305]: As far as I understand, interpAttrs.layout, interpAttrs.inPrc and interpAttrs.outPrc is not finalized yet at the time when this function is called. If jit kernel does not support fusion, here need an overkilled conditions, interpAttrs.mode == InterpolateMode::nearest  and getOriginalInputPrecisionAtPort(DATA_ID) == I8 or U8, no matter what layout it is. Ideally we should support fusion at all jitted kernels and remove these restriction.

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp
Comment (chenhu-wang) [inline src/plugins/intel_cpu/src/nodes/interpolate.cpp:1808]: could you please update code explanation comments to cover planar support by emulation case.

Comment (apavliuk-altran) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp]: Done

**Review by chenhu-wang — APPROVED** [review]

**Comment by apavliuk-altran** [comment]
@ceciliapeng2011 , @chenhu-wang , @dmitry-gorokhov 
If you don't have any more comments, could you please merge this PR after all checks will have passed?

**Comment by ceciliapeng2011** [comment]
@iefode and @jiwaszki, can you please take a look at this pr from ie-test and samples maintainers' perspective?

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/interpolate.cpp
Comment (iefode) [inline src/plugins/intel_cpu/tests/functional/single_layer_tests/interpolate.cpp:266]: quite strange hardcode

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/interpolate.cpp
Comment (iefode) [inline src/plugins/intel_cpu/tests/functional/single_layer_tests/interpolate.cpp:353]: look like to better use these configs in instansiation only for bf16

Comment (apavliuk-altran) [reply src/plugins/intel_cpu/tests/functional/single_layer_tests/interpolate.cpp:353]: Unfortunately, I haven't fully got your point.
The following key: `{InferenceEngine::PluginConfigParams::KEY_ENFORCE_BF16, InferenceEngine::PluginConfigParams::YES}` was present in original tests, and the only original precision used there was `ElementType::f32`
Do you suggest to completely remove this key until we add full BF16 support and corresponding tests?

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/interpolate.cpp
Comment (iefode) [reply src/plugins/intel_cpu/tests/functional/single_layer_tests/interpolate.cpp:266]: you check that it's `U8` and change the type after that without any reason.
Moreover, the comparation looks so strange (we compare the chars instead of element type). Why do you use such constractions?

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/interpolate.cpp
Comment (iefode) [reply src/plugins/intel_cpu/tests/functional/single_layer_tests/interpolate.cpp:353]: better to separate test instantiations for bf16 to standalone instance according using bf16 configs.

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/interpolate.cpp
Comment (apavliuk-altran) [reply src/plugins/intel_cpu/tests/functional/single_layer_tests/interpolate.cpp:266]: @iefode , the reason for this is to be consistent with the following workaround where exactly the same thing is done in existing code.
Could you please take a look at it?
https://github.com/openvinotoolkit/openvino/blob/c81fa39d3d40e305aa3783de2376ca3b3326533b/src/plugins/intel_cpu/src/node.cpp#L436
https://github.com/openvinotoolkit/openvino/blob/c81fa39d3d40e305aa3783de2376ca3b3326533b/src/plugins/intel_cpu/src/node.cpp#L442

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/interpolate.cpp
Comment (apavliuk-altran) [reply src/plugins/intel_cpu/tests/functional/single_layer_tests/interpolate.cpp]: Created separate function `filterAdditionalConfigNearestExt()` for Interpolate Nearest tests.
@iefode , could you please take a look?

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/interpolate.cpp
Comment (apavliuk-altran) [reply src/plugins/intel_cpu/tests/functional/single_layer_tests/interpolate.cpp:266]: @iefode , did you have an opportunity to take a look at the original workaround?

File: src/plugins/intel_cpu/src/nodes/interpolate.cpp
Comment (apavliuk-altran) [reply src/plugins/intel_cpu/src/nodes/interpolate.cpp]: This code is removed as we decided to support fusing for `int8` later

**Comment by akladiev** [comment]
This PR will be closed in 2 weeks in case of no activity.

**Comment by akladiev** [comment]
This PR was closed because it has been stalled for 2 week with no activity.

**Pull request closed without merging**
